transform-plugins/src/main/java/co/cask/hydrator/plugin/Decompressor.java

```diff
@@ -0,0 +1,315 @@
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xerial.snappy.Snappy;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * Decompreses the configured fields using the algorithms specified.
+ */
+@Plugin(type = "transform")
+@Name("Decompressor")
+@Description("Decompresses configured fields using the algorithms specified.")
+public final class Decompressor extends Transform<StructuredRecord, StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(Decompressor.class);
+  private final Config config;
+
+  // Output Schema associated with transform output.
+  private Schema outSchema;
+
+  // Output Field name to type map
+  private Map<String, Schema.Type> outSchemaMap = Maps.newHashMap();
+
+  // Map of field to decompressor type.
+  private final Map<String, DecompressorType> deCompMap = Maps.newTreeMap();
+
+  // This is used only for tests, otherwise this is being injected by the ingestion framework.
+  public Decompressor(Config config) {
+    this.config = config;
+  }
+
+  private void parseConfiguration(String config) throws IllegalArgumentException {
+    String[] mappings = config.split(",");
+    for (String mapping : mappings) {
+      String[] params = mapping.split(":");
+
+      // If format is not right, then we throw an exception.
+      if (params.length < 2) {
+        throw new IllegalArgumentException("Configuration " + mapping + " is in-correctly formed. " +
+                                             "Format should be <fieldname>:<decompressor-type>");
+      }
+
+      String field = params[0];
+      String type = params[1].toUpperCase();
+      DecompressorType cType = DecompressorType.NONE;
+
+      switch(type) {
+        case "SNAPPY":
+          cType = DecompressorType.SNAPPY;
+          break;
+
+        case "GZIP":
+          cType = DecompressorType.GZIP;
+          break;
+
+        case "ZIP":
+          cType = DecompressorType.ZIP;
+          break;
+
+        case "NONE":
+          cType = DecompressorType.NONE;
+          break;
+
+        default:
+          throw new IllegalArgumentException("Unknown decompressor type " + type + " found in mapping " + mapping);
+      }
+      if (deCompMap.containsKey(field)) {
+        throw new IllegalArgumentException("Field " + field + " already has decompressor set. Check the mapping.");
+      } else {
+        deCompMap.put(field, cType);
+      }
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
```

Comment:
I see, a little strange that we have an unchecked exception in the throws clause, but its unrelated to this PR then.  We should perhaps open a jira? @albertshau @gokulavasan ?